server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -177,8 +177,10 @@ private SortedKeyValueIterator<Key,Value> createIterator()
       files = reservation.getSecond();
     }
 
-    Collection<InterruptibleIterator> mapfiles =
+    List<InterruptibleIterator> mapfiles =
         fileManager.openFiles(files, scanParams.isIsolated(), samplerConfig);
+    // Randomize the ordering of files to avoid block cache contention on seeks
+    Collections.shuffle(mapfiles);
```

Comment:
I think you can use `tablet.getTableConfiguration()` to see if a shuffle property is set on the table. Another option, which is not currently wired up, would be to use an execution hint on a per-scan basis so that not all scans on the table are affected.